test/integration/sf.integration.ts

```diff
@@ -0,0 +1,209 @@
+  const versionToUpdateTo = '2.12.7-esm.0'
+  const channel = 'nightly'
+
+  const tarball =
+    process.platform === 'win32'
+      ? `https://developer.salesforce.com/media/salesforce-cli/sf/channels/${channel}/sf-win32-x64.tar.gz`
+      : `https://developer.salesforce.com/media/salesforce-cli/sf/channels/${channel}/sf-linux-x64.tar.gz`
+
+  before(async () => {
+    console.log('Setting up test environment...')
+
+    const {stdout} = await exec('npm view @salesforce/cli --json')
+    const distTags = JSON.parse(stdout)['dist-tags']
+
+    stableVersion = distTags.latest
+
+    testDir = await makeTestDir()
+    console.log(`Test directory: ${testDir}`)
+
+    dataDir = join(testDir, 'data')
+    cacheDir = join(testDir, 'cache')
+    configDir = join(testDir, 'config')
+
+    await mkdir(dataDir, {recursive: true})
+    await mkdir(cacheDir, {recursive: true})
+    await mkdir(configDir, {recursive: true})
+
+    process.env.SF_DATA_DIR = dataDir
+    process.env.SF_CACHE_DIR = cacheDir
+    process.env.SF_CONFIG_DIR = configDir
+
+    console.log('• data directory:', dataDir)
+    console.log('• cache directory:', cacheDir)
+    console.log('• config directory:', configDir)
+
+    const tarLocation = join(testDir, tarball.split('/').at(-1) ?? 'sf.tar.xz')
+    const extractedLocation = join(testDir, 'sf')
+    await mkdir(extractedLocation, {recursive: true})
+
+    await download(tarball, tarLocation)
+    const cmd =
+      process.platform === 'win32'
+        ? `tar -xf ${tarLocation} -C ${extractedLocation} --strip-components 1 --exclude node_modules/.bin`
+        : `tar -xf ${tarLocation} -C ${extractedLocation} --strip-components 1`
+
+    console.log(`Extracting ${tarLocation} to ${extractedLocation}`)
+    const extractResult = await exec(cmd, {cwd: testDir})
+    console.log(extractResult.stdout)
+    expect(extractResult.code).to.equal(0)
+    console.log('Success!')
+
+    console.log('Testing that installation was successful...')
+    // It's important to use run.js instead of sf - otherwise it will resolve to the global version of sf
+    sf = join(extractedLocation, 'bin', 'run.js')
+    // set the bin path so that plugin-update thinks this is updatable
+    // This would typically be set by the sf executable that's included in the tarball
+    // but since we're using bin/run.js to avoid the global sf, we need to set it manually
+    process.env.SF_BINPATH = sf
+
+    if (process.platform === 'win32') {
+      // append cmd /c to the command so that it can run on windows
+      sf = `cmd /c "node ${sf}"`
+    }
+
+    const versionResult = await exec(`${sf} version --json`)
+    console.log(versionResult.stdout)
+    expect(versionResult.code).to.equal(0)
+    initialVersion = JSON.parse(versionResult.stdout).cliVersion.replace('@salesforce/cli/', '')
+    console.log('Success!')
+
+    console.log('Linking plugin-update...')
+    // Running `plugins link` is very slow on github-action's windows runners. Writing this file
```

Comment:
will you change this since https://github.com/oclif/plugin-plugins/pull/679 merged?